File: software/examples/vbnet/ExampleCallback.vb
Imports System
Imports Tinkerforge

Module ExampleCallback
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your UV Light Bricklet

    ' Callback subroutine for UV light callback
    Sub UVLightCB(ByVal sender As BrickletUVLight, ByVal uvLight As Long)
        Console.WriteLine("UV Light: " + uvLight.ToString() + " µW/cm²")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim uvl As New BrickletUVLight(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register UV light callback to subroutine UVLightCB
        AddHandler uvl.UVLightCallback, AddressOf UVLightCB

        ' Set period for UV light callback to 1s (1000ms)
        ' Note: The UV light callback is only called every second
        '       if the UV light has changed since the last call!
        uvl.SetUVLightCallbackPeriod(1000)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletUVLight.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletUVLight;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change to your UID

// Callback function for UV light callback (parameter has unit µW/cm²)
function cb_uvLight($uv_light)
{
    echo "UV Light: $uv_light µW/cm²\n";
}

$ipcon = new IPConnection(); // Create IP connection
$uvl = new BrickletUVLight(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register UV light callback to function cb_uvLight
$uvl->registerCallback(BrickletUVLight::CALLBACK_UV_LIGHT, 'cb_uvLight');

// Set period for UV light callback to 1s (1000ms)
// Note: The UV light callback is only called every second
//       if the UV light has changed since the last call!
$uvl->setUVLightCallbackPeriod(1000);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
